### DISCUSSION TITLE
Webtop LSIO - how to export additional ports

### DISCUSSION THREAD

**Discussion by sloaiza86** [alexbelgium/hassio-addons#1331] [Q&A] [answered]
I need to expose an additional port for an app. Configuration only allows to select port for webui (default 3000). Any way to expose additional ports?

**Comment by alexbelgium** [comment]
Hi, which ports are you needing? Is it a recurrent one ?

There is no optimal solution for that : 
- either I enable host network, which means that the addon becomes fully opened in terms of ports exposure which decreases security, potentially causes issue if another addon uses the port 3000, but allows any new port to be mapped
- or I can map a specific other port (such as 3001, 3002) and you would use those specifically for your usages
- or you can open the ports manually using portainer, but this would need to be done at each restart and update

**Comment by sloaiza86** [comment]
Thanks! 

- Host network would be a good solution for me. I am needing ports 20 and 21 for ftp (I not able to use vsftp because I need to access a secondary hard disk and the add-on does not have such option) but as I am also needing Wireshark to capture some local traffic to troubleshoot another integration that could also help me with that task.
- I tried to map the port in portainer as follows (pretty new here to the container world, two months experience till now)

<img width="787" alt="image" src="https://github.com/alexbelgium/hassio-addons/assets/42628434/3f2649a0-f855-4add-b9c6-0723599a8a70">

Sadly, when I redeploy it, I got 404 error:

<img width="312" alt="image" src="https://github.com/alexbelgium/hassio-addons/assets/42628434/95d08f30-ff7f-445b-90a2-c6c9a2cf17d4">

Same happens if I try to change the network from Brigde to host.

**Comment by alexbelgium** [comment]
Hi, perhaps i'll still keep the specific ports mapping to avoid conflicts for other users.

I was thinking of perhaps mapping the full common drives but it might become confusing : 
PORT NO	PORT	PROTOCOL
21	FTP	TCP
22	SSH	TCP
23	TELNET	TCP
25	SMTP	TCP
53	DNS	TCP, UDP
67,68	DHCP	UDP
80	HTTP	TCP
110	POP3	TCP
111	Portmapper	TCP, UDP
123	NTP	UDP
137	NetBIOS	TCP, UDP
143	IMAP	TCP, UDP
161,162	SNMP	UDP
443	HTTPS	TCP

**Comment by sloaiza86** [comment, marked as answer]
What about a config line where the user can select the mapping required, similar to the line to select the Webui port?

**Reply by alexbelgium** [reply]
Indeed this is probably the optimal solution i'll implement it

**Comment by sloaiza86** [comment]
If 20 could be included for FTP data that would be great :)El 4/04/2024, a la(s) 3:46 a. m., Alexandre ***@***.***> escribió:﻿
Would this be ok ?
"ports": {
"21/tcp": null,
"22/tcp": null,
"23/tcp": null,
"25/tcp": null,
"3000/tcp": 3000,
"3001/tcp": null,
"3002/tcp": null
},
"ports_description": {
"21/tcp": "FTP",
"22/tcp": "SSH",
"23/tcp": "TELNET",
"25/tcp": "SMTP",
"3000/tcp": "web interface",
"3001/tcp": "custom port 1",
"3002/tcp": "custom port 2"
},

**Reply by alexbelgium** [reply]
ok I'll add it once the statistics daemons have run
